exercises/matthew_cumber/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,199 @@
+/**
+ * Skeleton main routine
+ */
+
+#include <iostream>
+#include <iomanip>
+#include <algorithm>
+#include <fstream>
+#include <string>
+#include <vector>
+#include <ctype.h>
+#include "UniqueWord.h"
+
+// Default path for output file reading files from
+std::string const PATH = "/Users/Matt/STFC Placement/C++ Introduction/newstarter/exercises/matthew_cumber/ex01_basics/";
+
+/** Tests if a word has already been found the increments it's count
+ * @param uniqueWords The vector of existing unique words found
+ * @param word The word that is being tested
+ * @return A boolean, true if a word already exists, false otherwise
+ */
+bool inUniqueWords(std::vector<UniqueWord> &uniqueWords,std::string word)
+{
+  // Loop unique words, if word exists increment it's count
+  for(auto &w : uniqueWords)
+  {
+    if(word == w.getWord())
+    {
+      w.incrementCount();
+      return true;
+    }
+  }
+
+  return false;
+}
+
+/** Tests if a letter is one of the predefined punctuation characters
+ * @param letter The letter which is being tested
+ * @return A boolean, true if a letter is a punctuation character, false otherwise
+ */
+bool isPunctuation(const char letter)
+{
+  for(int i = 0; i < PUNCTUATION_COUNT; ++i)
+  {
+    if(letter == PUNCTUATION[i]) return true;
+  }
+
+  return false;
+}
+
+/** Writes the unique words to an output file "UniqueWords.txt"
+ * @param uniqueWords The vector of unique words to be written to the file
+ * @return no return value
+ */
+void writeUniqueWordsToFile(std::vector<UniqueWord> const uniqueWords)
+{
+  // Try to opne the output file
+  std::ofstream file(PATH+"/src/output.txt");
+  if(!file.is_open())
+  {
+    std::cout << "Error opening file \"output.txt\", exiting ..." << std::endl;
+    return;
+  }
+
+  // Header of output file
+  file << "Word" << std::setw(FIELD_WIDTH) << "Usage" << std::endl << std::endl;
+
+  int fieldWidth = 10;
+
+  // To write words in descending order need to loop vector in reverse
+  for(auto w = uniqueWords.rbegin(); w != uniqueWords.rend(); ++w)
+  {
+    // Format the output to align count values
+    fieldWidth = FIELD_WIDTH-(int)w->getWord().length();
+    if(fieldWidth <= 0) file << w->getWord() << " " << w->getCount() << std::endl;
+    else file << w->getWord() << std::setw(fieldWidth) << w->getCount() << std::endl;
+  }
+
+  file.close();
+
+  std::cout << "Words successfuly saved in \"output.txt\"" << std::endl;
+
+  return;
+}
+
+int main(int argc, char const **argv)
+{
+  std::vector<UniqueWord> uniqueWords; // Stores words and counts as objects in a vector
+
+	// Check a file path has been provided as an argument on cl
+	if(argc < 2)
+	{
+		std::cout << "Please provide a path to the file to be read as a command line argument." << std::endl;
```

Comment:
Something funny has happened to the indenting here. Next time we get together it's worth discussing how to set up your development environment to apply clang code formatting that will do this for you. Generally speaking you can either apply it in the IDE or you can run a custom git command to apply the formatting before doing a commit